packages/react-components/react-color-picker-preview/library/src/utils/handleInputFieldEvents.ts

```diff
@@ -0,0 +1,16 @@
+import * as React from 'react';
+
+export const handleHexKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
```

Comment:
Thank you. Yeah, I was wondering where it's better to place it. I'll move to the stories

Reply:
I tried to simplify that but it didn't work. Because `tinycolor` make hex 6 symbols. For example if I type `f09` it'll add to the field additional symbols e.g. `ff0099`. That's why I need hex state which contains contents of a field.

Reply:
I apologize for any confusion. It would be more effective to use onChange with a controlled state instead of onKeyPress. Make sure to verify the value within setState, for instance: `setHex(oldHex => HEXT_REGEX.test(newHex) ? newHex : oldHex)`. This approach prevents the user from entering an incorrect value.